pkg/cloudprovider/external.go

```diff
@@ -17,12 +17,18 @@ const (
 // IsCloudProviderExternal is used to check whether external cloud provider settings should be used in a component.
 // It checks whether the ExternalCloudProvider feature gate is enabled and whether the ExternalCloudProvider feature
 // has been implemented for the platform.
-func IsCloudProviderExternal(platformType configv1.PlatformType, featureGate *configv1.FeatureGate) (bool, error) {
-	switch platformType {
+func IsCloudProviderExternal(infrastructureStatus configv1.InfrastructureStatus, featureGate *configv1.FeatureGate) (bool, error) {
+	switch infrastructureStatus.PlatformStatus.Type {
 	case configv1.AWSPlatformType,
 		configv1.OpenStackPlatformType:
 		// Platforms that are external based on feature gate presence
 		return isExternalFeatureGateEnabled(featureGate)
+	case configv1.AzurePlatformType:
+		if infrastructureStatus.PlatformStatus.Azure != nil &&
+			infrastructureStatus.PlatformStatus.Azure.CloudName == configv1.AzureStackCloud {
+			return true, nil
+		}
```

Comment:
For readability, what about extracting this into a little helper function?

```suggestion
		if isAzureStackHub(infrastructureStatus) {
			return true, nil
		}
```

and 
```
func isAzureStackHub(infrastructureStatus configv1.InfrastructureStatus) bool {
    return infrastructureStatus.PlatformStatus.Azure != nil &&
			infrastructureStatus.PlatformStatus.Azure.CloudName == configv1.AzureStackCloud
}